### ISSUE TITLE
Feature Request: Allow Transparency on BokehPass

### ISSUE THREAD

**Issue by planetMatrix** [mrdoob/three.js#19223]
Hi Dev's

Many thanks for this awesome Library and all the support system.

I have a feature request to make:

I am looking to get transparent background when rendering BokenPass.

@Mugen87 guided me but I guess this feature is not been implemented or in works.

https://discourse.threejs.org/t/is-there-a-way-to-set-clearpass-on-bokehpass/14615/2

I am on r115

Thanks again!

**Comment by Mugen87** [comment]
> I am looking to get transparent background when rendering BokenPass.

After some considerations it does not make sense to implement this feature request. An effect like DOF must be applied to the entire scene and background. If you support transparency with a DOF pass (and set the `alpha` property of the renderer to `true`), the composition of a HTML/CSS background and the post-processed scene will look wrong since you only blurred parts of the scene. The background color will bleed through and there is no way of fixing this.

I suggest you implement a background via `Scene.background` and do not touch the renderer's `alpha` parameter.

**Comment by Eviadc** [comment]
> After some considerations it does not make sense to implement this feature request.  

I think this feature could be somehow reasonable in particular cases.    

For example, I'm currently working on a project where I'd like to do some three js stuff on top of another canvas, which is rendered by a close-sourced 3D geographic map API. I have made some attempts with the `CSS3DRenderer` or using the map canvas as a `Texture`, but none of them worked out efficiently mainly because of the high texture resolution requirements to have a clear map image.  

At last, I had to render a three js scene with transparency, stacked right on top of the third-party map canvas, and then sync the coordinates of the two canvas in the render loop with a set of API. The map itself is flat though, but when viewing in 3D mode with a high polar angle, the distanced places near horizon will fade into a fog color.  
And after some tinkerings, it actually looks pretty weird for a three js mesh at a position where the map fades away, to have only fog on it but no blur at all. I've managed to create a bokeh effect and applied it to the scene, and the graphic looked just fine. To me, empirically, the one with bokeh effect looks more reasonable even not logically so.

In fact, even if I could integrate the map into the three js scene, I wouldn't want the dof effect to be applied to the map component. The information on the map should be either clear to be read, or too far to be able to see. Map information shouldn't be presented yet too blurry to read. In the meantime, I'd still like to have some neat dof effect on the objects above the map. It is, I'd say, logically false, but aesthetically 'correct', which means that this feature could be useful. Three js isn't a physics simulation library anyway.

I'm not sure if I have made my point clear, if not sorry for the bad language skill... I don't think I'm able to provide any demo or screenshots from the work before, but I can synthesis a simple demonstration of the idea with photoshop or something if necessary :D

**Comment by mrdoob** [comment]
TL;DR?

**Comment by Eviadc** [comment]
Well...

Yes, the feature might not make sense.

But still it could be aethetically useful,
since obviously we don't always want the physically correct graphics.

**Comment by mrdoob** [comment]
I think the forum is a better place to discuss this.

**Comment by Mugen87** [comment]
The problem is that if we change the code as suggested, inexperienced users will complain when the result don't look good. And there is no way of fixing this. Hence, I favor to not support this feature and let the user modifiy the sources if required.
